### PR TITLE
Add price range filter option

### LINKED ISSUE #24: Support to PriceRange filter

### PR DESCRIPTION
**Pull request by ClaudivanFilho**
#### What is the purpose of this pull request?
- Refactor code
- Add PriceRange filter

#### What problem is solving?
Fixes #24 

#### How should this be manually tested?
[workspace link](https://claudivan--storecomponents.myvtex.com/eletronicos/d?map=c%2Cc&rest=Smartphones)

#### Screenshots or example usage
screenshot from 2018-08-02 11-34-08

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/GalleryItem.js [hunk 1]
```diff
@@ -17,34 +17,16 @@ export default class GalleryItem extends Component {
...
-          .replace('https:', '')
-      }
-      newProduct.sku.referenceId = (newProduct.sku.referenceId &&
-        newProduct.sku.referenceId[0]) || {
-        Value: '',
-      }
-      delete newProduct.sku.sellers
-      delete newProduct.sku.images
-      delete newProduct.items
+    const normalizedProduct = { ...product }
```
Comment (brunojdo) [inline react/components/GalleryItem.js]: ✨

File: react/constants/SearchHelpers.js [hunk 2]
```diff
@@ -45,51 +13,115 @@ function getSpecificationFilterFromLink(link) {
...
+export function mountOptions(options, type, map, rest) {
+  const restMap = restMapped(rest, map)
+  return options.reduce((acc, opt) => {
+    const slug = getSlugFromLink(opt.Link)
+    let optMap = getMapByType(type)
+    if (type === 'SpecificationFilters') {
+      optMap = getSpecificationFilterFromLink(opt.Link)
+    }
+    const selected = restMap[slug && slug.toUpperCase()] === optMap
+    acc.push({
```
Comment (lucasecdb) [inline react/constants/SearchHelpers.js]: Don't use methods with side-effects inside the reduce.

File: react/constants/SearchHelpers.js [hunk 3]
```diff
@@ -45,51 +13,115 @@ function getSpecificationFilterFromLink(link) {
...
+      type,
+      slug,
+    })
+    return acc
+  }, [])
+}
+
+export function findInTree(tree, values, index) {
+  if (!(tree && tree.length && values.length)) return
+  for (let i = 0; i < tree.length; i++) {
```
Comment (lucasecdb) [inline react/constants/SearchHelpers.js]: Prefer the for...of loop instead of a traditional one.

File: react/constants/SearchHelpers.js [hunk 4]
```diff
@@ -45,51 +13,115 @@ function getSpecificationFilterFromLink(link) {
...
+/**
+ * Returns an object mapped by restValue and your mapValue.
+ * E.g.: rest='smartphones,lg' map='c,b' => { smartphones : 'c', lg: 'b' }
+ */
+function restMapped(rest, map) {
+  const restValues = (rest && rest.split(',')) || []
+  const mapValues = (map && map.split(',')) || []
+  const mapValuesSliced = mapValues.slice(restValues.length * -1)
+  return restValues.reduce((acc, value, index) => {
+    acc[value.toUpperCase()] = mapValuesSliced[index]
```
Comment (lucasecdb) [inline react/constants/SearchHelpers.js]: Do not mutate inside the reduce.

**Comment by ClaudivanFilho** [comment]
This PR solves #24

**Pull request merged**